docs: Authentication update (#7810)
googleapis/google-cloud-php: README.md

* Authenticate with `gcloud` to generate the credentials file.
* Instantiate a client.

### Installing the Google Cloud CLI
In order to generate our needed credentials file we need to authenticate to gcloud first.
Installation is handled differently depending on your platform. Here is a link to help you setup
the Google Cloud CLI:

https://cloud.google.com/sdk/docs/install

### Authenticate via the `gcloud` command
Once the Google Cloud CLI tools are installed it is required that we authenticate via the `gcloud`:
```shell
$ gcloud init
$ gcloud auth application-default login
```

This will create a local file in your system that the authentication library for our client will
read in order to make requests to the apis with those credentials. This file is located in different
place depending on your system.

Windows:
```
%APPDATA%\gcloud\application_default_credentials.json
```

Linux and MacOS:
```
$HOME/.config/gcloud/application_default_credentials.json
```

To read more about Authentication, see [AUTHENTICATION.md](AUTHENTICATION.md)

### Installing a client
Install the Google Translate client library with composer
```sh
composer install google/cloud-translate
```

> **Note**:For this example, we are using the Google Translate client library. Check the

Reviewer: nit: missing space

```suggestion
**Note**: For this example, we are using the Google Translate client library. Check the
```

$client = new TranslationServiceClient([]);

$request = new TranslateTextRequest();
$request->setParent('projects/<YOUR_PROJECT_ID>');
$request->setTargetLanguageCode('en-US');
$request->setContents(['こんにちは']);

// The request will contain the authentication token based on the default credentials file
$response = $client->translateText($request);

var_dump($response->getTranslations()[0]);
// {
//     ["translatedText"]=>
//     string(5) "Hello"
//     ["detectedLanguageCode"]=>
//     string(2) "ja"
// }

```

### Authentication

#### Note
This quickstart is built with local development in mind. The steps for deploying your project are
different depending on the environment you use. Here we provide some basic instruction in how to get
started with deployment of your project:

 * For applications running on Google Cloud Platform (for example, in Google Compute Engine, Google
   App Engine, Google Kubernetes Engine, and Google Cloud Functions), applications are automatically
   authenticated using the Metadata Server. For more information, see
   [About VM Metadata](https://cloud.google.com/compute/docs/metadata/overview)

>  * For applications running elsewhere, it is recommended to
>    [create a Service Account](https://cloud.google.com/iam/docs/service-accounts-create) and enable
>    [Workload Identity Federation](https://cloud.google.com/iam/docs/workload-identity-federation).
>  * It is also possible to authenticate using
>    [Service Account Keys](https://cloud.google.com/iam/docs/migrate-from-service-account-keys). Set
>    the `GOOGLE_APPLICATION_CREDENTIALS` environment variable pointing to your credentials file.
> 
> Authentication is handled by the client once the environment is configured. Once your credentials
> are created You just need to provide the authentication details when creating a client. Generally,
> authentication is accomplished using a Service Account. For more information on obtaining Service
> Account credentials see our
> [Authentication Guide](https://cloud.google.com/docs/authentication/production#manually).

Reviewer: ```suggestion
 * For applications running elsewhere, authentication is usually accomplished using a Service Account.
   For more information on obtaining Service Account credentials see our
[Authentication Guide](https://cloud.google.com/docs/authentication/production#manually). Set
   the `GOOGLE_APPLICATION_CREDENTIALS` environment variable pointing to your credentials file.
```

If you need support for other Google APIs, please check out the [Google APIs Client Library for PHP](https://github.com/google/google-api-php-client).

## Quickstart

> It is required to have the credentials configured to make requests to the api using the clients. In

Reviewer: I am not sure this first sentence is very useful. I would suggest removing it entirely.
```suggestion
It is required to have the credentials configured to make requests to the api using the clients. In
```

this guide we'll focus on how to configure your client for local development using the Google Cloud
CLI (`gcloud`).

### For local development:
* Install the Google Cloud CLI.
* Authenticate with `gcloud` to generate the credentials file.
* Instantiate a client.

### Installing the Google Cloud CLI
In order to generate our needed credentials file we need to authenticate to gcloud first.
Installation is handled differently depending on your platform. Here is a link to help you setup
the Google Cloud CLI:

https://cloud.google.com/sdk/docs/install

### Authenticate via the `gcloud` command
Once the Google Cloud CLI tools are installed it is required that we authenticate via the `gcloud`:
```shell
$ gcloud init
$ gcloud auth application-default login
```

This will create a local file in your system that the authentication library for our client will
read in order to make requests to the apis with those credentials. This file is located in different
place depending on your system.

Windows:
```
%APPDATA%\gcloud\application_default_credentials.json
```

Linux and MacOS:
```
$HOME/.config/gcloud/application_default_credentials.json
```

> 

Reviewer: can we add a `to read more about Authentication, see [AUTHENTICATION.md](AUTHENTICATION.md)`?

If you need support for other Google APIs, please check out the [Google APIs Client Library for PHP](https://github.com/google/google-api-php-client).

> ## Quickstart

Reviewer: I think the first thing a user should see is an explanation for how to install the packages. This could go even before the `Quickstart` header, so something like we had:

````
We recommend installing individual component packages. A list of available packages can be found on [Packagist](https://packagist.org/search/?q=google%2Fcloud-).

For example:
```sh
$ composer require google/cloud-storage
$ composer require google/cloud-bigquery
$ composer require google/cloud-datastore
```
````

And then the quickstart can follow after that.

In order to generate our needed credentials file we need to authenticate to gcloud first.
Installation is handled differently depending on your platform. Here is a link to help you setup
the Google Cloud CLI:

https://cloud.google.com/sdk/docs/install

### Authenticate via the `gcloud` command
Once the Google Cloud CLI tools are installed it is required that we authenticate via the `gcloud`:
```shell
$ gcloud init
$ gcloud auth application-default login
```

This will create a local file in your system that the authentication library for our client will
read in order to make requests to the apis with those credentials. This file is located in different
place depending on your system.

Windows:
```
%APPDATA%\gcloud\application_default_credentials.json
```

Linux and MacOS:
```
$HOME/.config/gcloud/application_default_credentials.json
```

To read more about Authentication, see [AUTHENTICATION.md](AUTHENTICATION.md)

### Instantiating the client
Once we have the `application_default_credentials.json` that we created on the previous step now we
can instantiate a client which internally using the Google Auth library will take that file and use
it to authenticate your requests:

> **Note**: You can install the Google Translate client library by running `composer install google/cloud-translate`.

Reviewer: Since this a quickstart, let's make this a required step (rather than a **NOTE**):

````suggestion
Install the Google Translate client library with composer

```sh
composer install google/cloud-translate
```
````

See the [README.md](README.md) for further instructions, or browse

> [the complete list of packages](https://cloud.google.com/php/docs/reference/).

Reviewer: We are already in the readme, so no reason to link to the README.

```php
require_once 'vendor/autoload.php';

use Google\Cloud\Translate\V3\Client\TranslationServiceClient;
use Google\Cloud\Translate\V3\TranslateTextRequest;

// Instantiating the client gathers the credentials from the `application_default_credentials.json`
$client = new TranslationServiceClient([]);

$request = new TranslateTextRequest();
$request->setParent('projects/<YOUR_PROJECT_ID>');
$request->setTargetLanguageCode('en-US');
$request->setContents(['こんにちは']);

// The request will contain the authentication token based on the default credentials file
$response = $client->translateText($request);

var_dump($response->getTranslations()[0]);
// {
//     ["translatedText"]=>
//     string(5) "Hello"
//     ["detectedLanguageCode"]=>
//     string(2) "ja"
// }

```

> This quicksetup is built with local development in mind, the steps for deploying your project are

Reviewer: Did you mean `quickstart` ?

> This quicksetup is built with local development in mind, the steps for deploying your project are

Reviewer: Should be a period, not a comma
```suggestion
This quicksetup is built with local development in mind. The steps for deploying your project are
```

> different depending on the environment you use. Here we provide some basic instruction in how to get
> started with deployment of your project:
> 
> ### GCP deployment:
> * Use the metadata server instead of a credentials file.
>     - More information about the usage of the metadata server
>       [here](https://cloud.google.com/compute/docs/metadata/overview)
> * Instantiate a client.
> 
> ### Server deployment:
> * Generate a credentials file.
>     - We recommend using a Service Account instead of the personal account credentials. More
>       information [here](https://cloud.google.com/iam/docs/service-accounts-create)
> * Set the GOOGLE_APPLICATION_CREDENTIALS environment variable pointing to your credentials file.
> * Instantiate a client.

Reviewer: This section needs to be reworked. I would simplfy it by saying something like this instead:

````suggestion
different depending on the environment you use. 

 * For applications running on Google Cloud Platform (for example, in Google Compute Engine, Google App Engine, Google Kubernetes Engine, and Google Cloud Functions), applications are automatically authenticated using the Metadata Server. For more information, see [About VM Metadata](https://cloud.google.com/compute/docs/metadata/overview)
 * For applications running elsewhere, it is recommended to [create a Service Account](https://cloud.google.com/iam/docs/service-accounts-create) and enable [Workload Identity Federation](https://cloud.google.com/iam/docs/workload-identity-federation).
 * It is also possible to authenticate using [Service Account Keys](https://cloud.google.com/iam/docs/migrate-from-service-account-keys). Set the `GOOGLE_APPLICATION_CREDENTIALS` environment variable pointing to your credentials file.

````

    'keyFile' => json_decode(file_get_contents('/path/to/keyfile.json'), true)
]);
```
A list of clients that accept these parameters are:
- [Bigtable](https://github.com/googleapis/google-cloud-php-bigtable)
- [Spanner](https://github.com/googleapis/google-cloud-php-spanner)
- [Firestore](https://github.com/googleapis/google-cloud-php-firestore)
- [Datastore](https://github.com/googleapis/google-cloud-php-datastore)

> - [Pubsub](https://github.com/googleapis/google-cloud-php-pubsub)

Reviewer: We also should add Debugger, Logging, Trace, Translate, BigQuery, and Storage
